File: onlinejudge/leetcode/2965.rb
def find_missing_and_repeated_values(grid)
  n = grid.length ** 2
  set = Set.new
  answer = [0, 0]

  grid.each do |row|
    row.each do |num|
      if set.include?(num)
        answer[0] = num
        break
      end
      set.add(num)
    end
  end

  (1..n).each do |i|
    unless set.include?(i)
      answer[1] = i
      break
    end
  end

  answer
end

File: onlinejudge/leetcode/2965.go

func findMissingAndRepeatedValues(grid [][]int) [2]int {
	n := len(grid) * len(grid)
	set := make(map[int]bool)
	answer := [2]int{}

	for _, row := range grid {
		for _, num := range row {
			if set[num] {
				answer[0] = num
				break
			}
			set[num] = true
		}
	}

	for i := 1; i <= n; i++ {
		if !set[i] {
			answer[1] = i
			break
		}
	}

	return answer
}
